scripts/backport.py

```diff
@@ -0,0 +1,261 @@
+        )
+        continue
+
+    if "no-backport" in issue_labels:
+        print(
+            f"{commit_sha[:9]}\tskip\tissue marked as 'no-backport'\t{issue.number}\t{issue.title}"
+        )
+        continue
+
+    print(f"{commit_sha[:9]}\tbackport\t\t{commit_sha}\t{commit_title}")
+
+    # Remember the PR and the corresponding resulting commit in main.
+    if pull.number not in prs_to_backport:
+        prs_to_backport[pull.number] = pull
+        pr_commits[pull.number] = [pygithub_commit]
+    else:
+        # We're traversing the history backwards, and want to have the list of
+        # commits in forward order.
+        pr_commits[pull.number].insert(0, pygithub_commit)
+
+
+# Now, go over the list of PRs that we have collected, and try to backport
+# each of them.
+for original_pr in prs_to_backport.values():
+    backport_branch = f"backport/{backport_target}/{original_pr.number}"
+
+    # If there is already a backport branch for this PR, just skip it.
+    if (
+        subprocess.run(
+            f"git rev-parse {target_remote}/{backport_branch} > /dev/null 2>&1",
+            shell=True,
+        ).returncode
+        == 0
+    ):
+        print(
+            f'Backport branch {backport_branch} for PR #{original_pr.number}: "{original_pr.title}" already exists. Skipping.'
+        )
+        continue
+
+    # Try to cherry-pick the commits.
+    subprocess.run(
+        f"git checkout --quiet --detach {source_remote}/{backport_target} > /dev/null",
+        shell=True,
+    ).check_returncode()
+
+    commit_list = " ".join([commit.sha for commit in pr_commits[original_pr.number]])
+
+    if subprocess.run(
+        # f"git cherry-pick -m 1 -x {commit_list}",
+        f"git cherry-pick -m 1 -x {commit_list} > /dev/null 2>&1",
+        shell=True,
+    ).returncode:
+        subprocess.run(
+            f"git cherry-pick --abort > /dev/null", shell=True
+        ).check_returncode()
+        print(
+            f'Conflict while backporting pull request #{original_pr.number}: "{original_pr.title}"'
+        )
+        original_pr.add_to_labels("pr-auto-backport-failed")
+        continue
+
+    # Push the branch and create the backport PR
+    subprocess.run(
+        f"git push {target_remote} @:refs/heads/{backport_branch} > /dev/null",
+        shell=True,
+    ).check_returncode()
+
+    backport_pr = target_repo.create_pull(
+        title=f"""
+            Backport to {backport_target}: #{original_pr.number}: {original_pr.title}
+            Merge it after all the relevant CI checks pass.
+            If this fix should not be backported, or will be backported manually, just close this PR.
+        """,
+        body=f"This is an automated backport of #{original_pr.number}: {original_pr.title}",
+        head=backport_branch,
+        base=backport_target,
+    )
+    backport_pr.add_to_labels("pr-backport")
+    backport_pr.add_to_assignees(original_pr.user.login)
```

Comment:
Was this line added after you opened the test backport PRs? The open PRs are not assigned to anybody?

Reply:
A nice to have feature: is it be possible to enable auto merge for the PR? I mean, if it can be built and all tests are successful, it could be merged.

Reply:
> Was this line added after you opened the test backport PRs? The open PRs are not assigned to anybody?

Yes, I fixed this later.

> A nice to have feature: is it be possible to enable auto merge for the PR? I mean, if it can be built and all tests are successful, it could be merged.

Unfortunately, I didn't find a way to do this through Github API. I think to later add a "phase 3" to this script, which would just go through the open PRs marked as 'pr-backport' with green checks, and merges them. Probably in the next PR, because the API for checks seems to be tricky.

Reply:
Maybe this helps: https://docs.github.com/en/graphql/reference/mutations#enablepullrequestautomerge

Reply:
Thanks, that's why I didn't find it, it's the scary GraphQL API :) I try to stick to the RESTful v3 where possible. I'll take a look at it.

Reply:
Although it got me thinking, auto-merge would require us setting the required checks for the release branches, do we want this? Currently we have none. Maybe it'll make things less convenient.

Reply:
Good point... On the other hand, do we want to be able to merge PRs if tests fail? Maybe just mark the tests as required and not require reviews in that branch?

Reply:
OK, let's try it, I added some branch-protection rules (regression and up/downgrade tests like in main), and the code to enable auto-merge.

Reply:
Github graphql returns status 200 on failed requests, and apparently this is a "common convention" :facepalm: